src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -33,13 +35,35 @@ impl KbcInterface for Kbc {
         Err(anyhow!("Decrypt Payload API of this KBC is unimplemented."))
     }
 
-    async fn get_resource(&mut self, _description: &str) -> Result<Vec<u8>> {
-        Err(anyhow!("Unimplemented"))
+    #[allow(unused_assignments)]
+    async fn get_resource(&mut self, description: &str) -> Result<Vec<u8>> {
+        let desc: ResourceDescription = serde_json::from_str::<ResourceDescription>(description)?;
+        let mut resource_url = String::default();
+        let resource_type = desc
```

Comment:
I suggest to impl `TryInto<String>` for `ResourceDescription`. Then it will be convenient whenever it needs to convert a `ResourceDescription` into a `String`

Reply:
Good suggestion, but it shouldn't be this PR's work.